java/client/src/org/openqa/selenium/io/TemporaryFilesystem.java

```diff
@@ -145,6 +145,10 @@ boolean shouldReap() {
   }
 
   public boolean deleteBaseDir() {
-    return baseDir.delete();
+    boolean wasDeleted = baseDir.delete();
+    if (wasDeleted) {
```

Comment:
I don't think we need to necessarily check if that returned true or not, we should just remove the shutdown hook. Since nothing would check or do anything with this flag anyways.